sympy/concrete/summations.py

```diff
@@ -988,6 +988,24 @@ def eval_sum_symbolic(f, limits):
 
         r = gosper_sum(f, (i, a, b))
 
+        # to find Sum at points where denominator approaches to zero
+        from sympy import together, fraction, Mul, Add, ordered
+        if isinstance(r, (Mul,Add)) and not(any(x in r.atoms(Symbol) for x in limits)):
+            var = ordered(r.atoms(Symbol))
+            _, denom = fraction(together(r))
+            if any(x in denom.atoms(Symbol) for x in var):
+                cond = []
+                for v in var:
+                    for s in solve(denom, v):
+                        m = Eq(v, s)
+                        if not(m == False and m == True):
+                            cond.append(m)
+                pic = []
```

Comment:
Although you probably pronounce this as "piece" in your mind, many will read this as "pick". You could make it more generic by just writing

```
args = [(Sum(f_orig.subs(*m.args), limits).doit(), m) for m in cond]
args.append((r, True))
r = Piecewise(*args)
```

I don't do much with this area of code, however, and can't really comment on the merits of what you are doing, e.g. using `doit` on the expressions you are putting into the Piecewise. I also know that we are trying to move away from solve towards solveset. But then you should watch for unexpected output, e.g. non-FiniteSet ?